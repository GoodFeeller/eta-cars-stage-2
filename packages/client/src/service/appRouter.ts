import { createTRPCProxyClient, httpBatchLink } from '@trpc/client'
import { appRouterType } from '../../../server/src/appRouter'
const trpc = createTRPCProxyClient<appRouterType>({
  links: [
    httpBatchLink({
      url: 'https://eta-cars-internship-api.onrender.com/trpc',
    }),
  ],
})

export default trpc
