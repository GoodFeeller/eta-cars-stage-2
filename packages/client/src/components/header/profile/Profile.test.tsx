import { render, screen } from '@testing-library/react'
import Profile from './Profile'
import useProfile from '../../../hooks/useProfile'
import { IProfileTotal } from '../../../service/ProfileTypes'

describe('Profile tests', () => {
  test('Full negative profile snapshot', () => {
    const profile: IProfileTotal = {
      totalPrice: 10000,
      id: ['bitcoin'],
    }
    const nowPrice = 9000
    jest.mock('../../../hooks/useProfile', () => {
      const original = jest.requireActual('../../../hooks/useProfile')
      return {
        __esModule: true,
        ...original,
        default: () => {
          return { profile, nowPrice }
        },
      }
    })
    render(<Profile />)
    const profileBox = screen.getByTestId('profile')
    expect(profileBox).toMatchSnapshot()
  })
  test('Full positive profile snapshot', () => {
    const profile: IProfileTotal = {
      totalPrice: 10000,
      id: ['bitcoin'],
    }
    const nowPrice = 11000
    jest.mock('../../../hooks/useProfile', () => {
      const original = jest.requireActual('../../../hooks/useProfile')
      return {
        __esModule: true,
        ...original,
        default: () => {
          return { profile, nowPrice }
        },
      }
    })
    render(<Profile />)
    const profileBox = screen.getByTestId('profile')
    expect(profileBox).toMatchSnapshot()
  })
  test('Empty profile snapshot', () => {
    const profile: IProfileTotal = {
      totalPrice: 0,
      id: [],
    }
    const nowPrice = 0
    jest.mock('../../../hooks/useProfile', () => {
      const original = jest.requireActual('../../../hooks/useProfile')
      return {
        __esModule: true,
        ...original,
        default: () => {
          return { profile, nowPrice }
        },
      }
    })
    render(<Profile />)
    const profileBox = screen.getByTestId('profile')
    expect(profileBox).toMatchSnapshot()
  })
})
