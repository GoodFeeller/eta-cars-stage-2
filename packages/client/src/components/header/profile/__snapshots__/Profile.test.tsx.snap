// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests Empty profile snapshot 1`] = `
<div
  class="profile"
  data-testid="profile"
>
  <span>
    Profile
  </span>
  <span>
    TotalPrice:
     
    $0.00
  </span>
  <span
    style="color: black;"
  >
    $0.00
  </span>
  <span
    style="color: black;"
  >
    0%
  </span>
</div>
`;

exports[`Profile tests Full negative profile snapshot 1`] = `
<div
  class="profile"
  data-testid="profile"
>
  <span>
    Profile
  </span>
  <span>
    TotalPrice:
     
    $0.00
  </span>
  <span
    style="color: black;"
  >
    $0.00
  </span>
  <span
    style="color: black;"
  >
    0%
  </span>
</div>
`;

exports[`Profile tests Full positive profile snapshot 1`] = `
<div
  class="profile"
  data-testid="profile"
>
  <span>
    Profile
  </span>
  <span>
    TotalPrice:
     
    $0.00
  </span>
  <span
    style="color: black;"
  >
    $0.00
  </span>
  <span
    style="color: black;"
  >
    0%
  </span>
</div>
`;
